### ISSUE TITLE
Blocks even when turned off.

### ISSUE THREAD

**Issue by CTimmerman** [uBlockOrigin/uBlock-issues#799]
### Prerequisites

- [X] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [X] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [X] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [ ] uBlock Origin is the only extension
    - [ ] uBlock Origin with default lists/settings
    - [ ] using a new, unmodified browser profile
- [ ] I am running the latest version of uBlock Origin
- [ ] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

I tried to download an ad, but Chrome says the download failed due to it being blocked. Odd, since the only blocker i have installed is uBlock Origin and that's turned off according to its icon. Disabling it in the Chrome settings allowed me to download the ad.

### A specific URL where the issue occurs

[A specific URL is MANDATORY for issue happening on a web page, even if it happens "everywhere"]
You asked for it:

`https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4`

^ Not Safe For Prudes

### Steps to Reproduce

1. Visit Gelbooru with uBlock Origin set to "off" via its menu.
2. Notice the amusing ads.
3. Try to save said amusing ads.

### Expected behavior:

Save the video.

### Actual behavior:

Download fails because "Blocked".

### Your environment

* uBlock Origin version: 1.23.0
* Browser Name and version: Chrome 78.0.3904.108 (Officiële build) (64-bits)
* Operating System and version: Windows 10 Home 1903 build 18362.476

**Comment by uBlock-user** [comment]
@gwarser

**Comment by gwarser** [comment]
Tabless request? With context rewritten to 'self'?

**Comment by gwarser** [comment]
`behind-the-scene`

Chrome 1.23:

| | | | | | | |
|:--- |:--- |:--- |:--- |:--- |:--- |:--- |
| +0 | \|\|contentabc.com^$3p | -- | behind-the-scene | 0,3 | media | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |

Chrome 1.24.3b1:

| | | | | | | |
|:--- |:--- |:--- |:--- |:--- |:--- |:--- |
| +0 |  |  | hw-cdn2.contentabc.com | 0,1 | media | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |

Already fixed, just wait for update.

**Comment by gwarser** [comment]
Firefox70, 1.23 and 1.24, blocked:

![image](https://user-images.githubusercontent.com/886325/69863637-9e1baa00-129d-11ea-87a6-3437e2cc09e6.png)

| | | | | | | |
|:--- |:--- |:--- |:--- |:--- |:--- |:--- |
| +0 | \|\|contentabc.com^$3p | -- | github.com | 0,3 | other | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |

Whitelisted:

| | | | | | | |
|:--- |:--- |:--- |:--- |:--- |:--- |:--- |
| +1 |  |  | github.com | 0,3 | other | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |

**Comment by gwarser** [comment]
Related https://github.com/gorhill/uBlock/commit/dd2a9faa4c024b8f87fa09b42ed542521c69ddc9

**Comment by uBlock-user** [comment]
Can't reproduce it in Nightly

**Comment by gorhill** [comment]
So is this something that was fixed already?

**Comment by gwarser** [comment]
Most likely. Waiting for confirmation from OP or link to original source page.

**Comment by CTimmerman** [comment]
> Most likely. Waiting for confirmation from OP or link to original source page.

1. Re-enable 1.23 and note it's still turned "off".
2. Go to https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4
3. Save that video which plays just fine, i.e. not blocked at all.
4. See the download fail because "Blocked".
5. Disable uBlock Origin and consider NoScript if you get too many popups.

**Comment by uBlock-user** [comment]
Where did you find that video on ? Link that specific page.

**Comment by gwarser** [comment]
Ok ,so when accessing mp4 directly:

1.23, `+0` entry without whitelisting `contentabc.com`:

| | | | | | | |
|:--- |:--- |:--- |:--- |:--- |:--- |:--- |
| +22 | \|\|contentabc.com^$3p | -- | behind-the-scene | 0,3 | media | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |
| +19 | whitelisted | ++ | hw-cdn2.contentabc.com | 1 | doc | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |
| +0 | \|\|contentabc.com^$3p | -- | behind-the-scene | 0,3 | media | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |

1.24.3b1:

| | | | | | | |
|:--- |:--- |:--- |:--- |:--- |:--- |:--- |
| +27 |  |  | hw-cdn2.contentabc.com | 0,1 | media | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |
| +24 | whitelisted | ++ | hw-cdn2.contentabc.com | 1 | doc | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |
| +10 |  |  | hw-cdn2.contentabc.com | 0,1 | media | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |
| +6 |  |  | hw-cdn2.contentabc.com | 1 | media | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |
| +0 |  |  | hw-cdn2.contentabc.com | 1 | doc | `https://hw-cdn2.contentabc.com/a7/creatives/1/77/805214/912754/912754_video_low.mp4` |


Fixed.

**Comment by CTimmerman** [comment]
> Where did you find that video on ? Link that specific page.

The specific page is lost and irrelevant. Enable porn on the listed site and any page will do, e.g. https://gelbooru.com/index.php?page=post&s=list&tags=blue+sky+cloud+1girl

**Comment by gwarser** [comment]
Context is relevant:

- viewing on gelbooru - whitelisted
- saving directly from gelbooru - context is gelbooru - connection should be whitelisted
- saving from ads iframe on gelbooru:
	- when displayed on page it's whitelisted
	- when downloading by right clicking, context is set to ads frame , so not whitelisted:
		- can be blocked because third-party to iframe (top frame? there was some shuffling here depending on filter type)
		- can be blocked by incorrect 3p `b-t-s` context
- opening in new tab and then downloading: 
	- no longer whitelisted because `contentabc.com` is not related in any way to gelbooru
	- can be opened for viewing because no "document" filters exist for `contentabc`
	- but when downloading type changes to "other" or "media" depending on server response:
		- can be incorrectly third-party because of incorrect `b-t-s` context

Pretty complicated. I'm not even sure these are all cases.

Note: videos are covered by image, I removed them in inspector to access video context menu.

**Comment by gorhill** [comment]
> is there any way to pass "whitelisted" state to content downloaded from whitelisted page?

uBO deal with network requests and the browser does not tell uBO that a specific network request is the result of a "download" operation. In any case the specific issue is solved here by commit <https://github.com/gorhill/uBlock/commit/dd2a9faa4c024b8f87fa09b42ed542521c69ddc9>? If ever there is a different case then a new issue should be opened -- assuming it's not working as intended.

**Comment by gwarser** [comment]
Interesting. These download requests are only visible in uBO, neither ABP, AdBlock or even DevTools show them.

>uBO deals with network requests and the browser does not tell uBO that a specific network request is the result of a "download" operation. 

Maybe `behind-the-scene` is fixed and tells exactly this?

**Comment by CTimmerman** [comment]
It something is blocked, it shouldn't load at all; conversely, if something is not blocked, the user should be allowed to save it.

**Comment by uBlock-user** [comment]
The fix is in the dev build if you want to try.
